### ISSUE TITLE
Whats going on with Calendar+ development?

### ISSUE THREAD

**Issue by farson2003** [libasys/calendarplus#88]
Hello @libasys 
Not sure what is going on here, but while @libasys seems to have stepped back from showing himself here for some weeks, @jancborchardt and @LukasReschke seem to have flooded calendar+ with issues...

We would like to start using calendar+ in productive environments... hence it would be good to know...
Also related to this open issue:https://github.com/libasys/calendarplus/issues/53

Thanks
##

**Comment by jancborchardt** [comment]
I also wonder about this … @DeepDiver1975 and I sat together with @libasys a few weeks ago to decide if Calendar+ is the way to go for ownCloud’s official calendar from ownCloud 8.2 on. That’s why @LukasReschke and me reviewed it and opened so many issues.

Unfortunately it seems though that there has not been much activity from @libasys recently, and the apps are also still very buggy and have lots of design problems.
We again see the danger of having only one maintainer … same problem as with the current calendar – one person can just not invest enough time to keep an app stable (or even multiple ones as @libasys attempts). There needs to be a community around an app, so we need to collaborate.

**Comment by farson2003** [comment]
@jancborchardt Thank you for your thoughts.

Please allow me to speak up:

A respectable amount of work went into both apps "Calendar" and "Calendar+". Users are grateful for the superb volunteer work. 
As we all know a calendar application is one of the most essential components of any Owncloud instance.

However, for both administrators and endusers there is nothing worse than uncertainty with regards to future app maintenance.

Please, @jancborchardt , @libasys , @karlitschek , @georgehrke , @tanghus , @bartv2,@LukasReschke , @DeepDiver1975 , do reach a compromise on how to proceed with such essential calendar application.

While Calendar+ offers great new features, it is difficult to be maintained by one person only.

Looking at the amount of work and limited ressources it seems reasonable to reach a compromise and focus the energy of all developer into one app, whichever that may be (that's only my personal opinion, though :-) ).

At his very moment as an "enduser" and administrator one now is left with great uncertainty which Calendar app to install, Calendar, or Calendar+, as their maintenance appears uncertain. 
Therefore a mutual statement form all developers involved, possibly also backed by the project leader, @karlitschek, would be helpful.

Thank you

**Comment by libasys** [comment]
@jancborchardt it isn't still buggy, i fixed hundreds of bugs and added several big features like public sharing, subscribing to calendar etc! At the moment I have serveral other projects where i have to work for! So stay cool and by the way the normal calendar app aren't developed for a long distance (1-2 years) only bugfixes!

**Comment by LukasReschke** [comment]
>  @jancborchardt and @LukasReschke seem to have flooded calendar+ with issues...

How is https://github.com/libasys/calendarplus/issues/created_by/LukasReschke supposed to be flooding? These are barely comments on coding best practises which are very highly recommended (others would say: required) if this is going to be an official replacement.

**Comment by libasys** [comment]
@LukasReschke your comments every time welcome!

**Comment by jancborchardt** [comment]
@LukasReschke I guess @farson2003 didn’t mean this in a negative way, maybe? ;)

**Comment by jancborchardt** [comment]
@libasys sorry – I didn’t mean to come off as negative. It’s just a very frustrating overall situation, and we really need to work towards getting more people involved with this new Calendar.

@DeepDiver1975 @libasys maybe it’s time to move the Calendar+ and Contacts+ repositories over to the ownCloud organization so we can work more effectively on them? If they are to become the new official apps, we can’t just have one maintainer.

**Comment by farson2003** [comment]
@LukasReschke @libasys @DeepDiver1975
I stronlgy believe Calendar+ is a wonderful step forward, and I have great respect for all the hard work that went into it!
 I could put the hand on my heart being certain many users would prefer Calendar+, and many would like it to become official (see also my post here: https://github.com/libasys/calendarplus/issues/53)

So, @libasys , I definetely was not negative. 
I just try to be a voice of many admins who see the improvements but also associated uncertainties...(but maybe I am over cautious ? :-) 

Please consider.

**Comment by jancborchardt** [comment]
I think all of us discussing here, opening issues and trying to move Calendar+ forward means we all care a lot about it. :) All constructive feedback is meant in a positive way, even though communication through text can sometimes lead to misunderstanding.

**Comment by karlitschek** [comment]
I agree with @jancborchardt Calendar plus is great and i hope that more users will use it in the future and more contributors will help to push this forward.
Speaking about this. Are there any plans to move the galleryplus repo into the owncloud organisations so that more developers see what is going on an can contribute? Doesn't has to happen but would be cool.
Hint: This doesn't mean that we will replace the other calendar repo/app. I'm happy to have two different approaches in parallel.

**Comment by jancborchardt** [comment]
@karlitschek yup, that’s what I was asking above also. :)

> @DeepDiver1975 @libasys maybe it’s time to move the Calendar+ and Contacts+ repositories over to the ownCloud organization so we can work more effectively on them? If they are to become the new official apps, we can’t just have one maintainer.

**Comment by DeepDiver1975** [comment]
So move or not?

calendar+, contacts+ and tasky+ are highly integrated - If so we move all or nothing

@libasys this is basically our call if you are willing to move the repos over

**Comment by libasys** [comment]
yes i want ( sounds a marriage ;) ), and what i have todo?

**Comment by DeepDiver1975** [comment]
> yes i want ( sounds a marriage ;) ), and what i have todo?

:bride_with_veil:

**Comment by DeepDiver1975** [comment]
so you have to transfer the repos to me and I'll move them forward to the ownCloud orga

**Comment by DeepDiver1975** [comment]
@karlitschek @jancborchardt agreement to move all 3 apps to the ownCloud orga?

**Comment by jancborchardt** [comment]
3? You mean calendarplus, contactsplus, that’s 2.

**Comment by DeepDiver1975** [comment]
> calendar+, contacts+ and tasky+ are highly integrated - If so we move all or nothing

@jancborchardt

**Comment by jancborchardt** [comment]
If you refer to tasksplus with the third one I would say no. https://github.com/owncloud/tasks is a great app, it’s well maintained and @raimund-schluessler and @henni do a great job. It’s not in need of any replacement.

**Comment by jancborchardt** [comment]
That is, I suggest we move to calendarplus and contactsplus and collaborate on that. And for tasks I suggest @libasys you collaborate on https://github.com/owncloud/tasks :)

**Comment by DeepDiver1975** [comment]
> calendar+, contacts+ and tasky+ are highly integrated - If so we move all or nothing

it's all or nothing from my pov - they belong together

**Comment by DeepDiver1975** [comment]
> And for tasks I suggest @libasys you collaborate on https://github.com/owncloud/tasks :)

not possible because they have it's own backends - integration is not possible

**Comment by DeepDiver1975** [comment]
This is basically the reason why splitting front end and backend into separate parts - https://mailman.owncloud.org/pipermail/devel/2015-July/001416.html

**Comment by libasys** [comment]
I transfer the repo to DeepDiver1975, ok? it is the danger zone ^^

**Comment by DeepDiver1975** [comment]
> I transfer the repo to DeepDiver1975, ok? it is the danger zone ^^

I first want to have a confirmation on what being moved

**Comment by raimund-schluessler** [comment]
@DeepDiver1975 You are right, https://github.com/owncloud/tasks currently depends on the calendar app. I started removing all calls to `OC_Calendar_**`-functions (although the work on this is not very advanced yet) which should make maintaining the app easier. Also it should allow an easier migration to a different backend, may it be a CalDAV backend in core as described in https://mailman.owncloud.org/pipermail/devel/2015-July/001416.html (which I would really like to see) or simply a different calendar app.

However, before I spend any work on migrating the tasks app to whatever backend, we have to be sure that we will stick with this. Anyway, I do not plan to quit on developing the app, as I also use it myself ;).

**Comment by raimund-schluessler** [comment]
@DeepDiver1975 @jancborchardt @libasys This maybe can be discussed at the owncloud conference?

**Comment by Henni** [comment]
awesome to see this happening!

@raimund-schluessler is currently removing the Calendar dependencies from the tasks app.
PR for reference owncloud/tasks#228

**Comment by raimund-schluessler** [comment]
> @raimund-schluessler is currently removing the Calendar dependencies from the tasks app

As far as this is possible. We of course still need the respective database tables and CalDAV access/syncing will still be handled by the calendar app (not reasonable to implement two CalDAV entry points, just to make it run "standalone").

**Comment by karlitschek** [comment]
moving an appnin the owncloud organisation has nothing to do with 'replacing' an other app. there is nonproblem at all with having two awesome tasks app in the organisation.

**Comment by libasys** [comment]
@raimund-schluessler keep up your good work and develope on it! Long time ago it was the only logic step for me, to write an own task app, cause i changed up calendar to it own namespace and worked with more actual sabre dav libs! Great idea to split the backend all to the tasks app, one small hint, use on your db mapper class const for tablenames so it is much easier for you to switch from current calendar table to another!

**Comment by jancborchardt** [comment]
> moving an appnin the owncloud organisation has nothing to do with 'replacing' an other app. there is nonproblem at all with having two awesome tasks app in the organisation.

@karlitschek Sorry but I disagree. We have to focus our efforts here, and we already spread it way too thin. The existing Tasks app conforms to our styles better, works and is maintained by 2 people, so we should keep that as the default. For Calendar and Contacts we should focus on the »plus« versions. We’re just discussing how we can make better collaboration happen.

**Comment by DeepDiver1975** [comment]
> We’re just discussing how we can make better collaboration happen.

Not possible since they have no tech base in common - pretty easy.

**Comment by jancborchardt** [comment]
Ok, so how do we proceed here? Prioritize the splitting of the sync backend, and then continue after that?

The Contacts app is independent of this so we could already move that. However, Contacts is used by some other apps (Mail and Maps for example) so we should make sure the calls used there are compatible.

**Comment by DeepDiver1975** [comment]
> Ok, so how do we proceed here? Prioritize the splitting of the sync backend, and then continue after that?

This will not happen in the next 6 month - this might be a solution for OC 9.1+

For 8.2+ I only see to focus on the tree plus apps and move them forward.

**Comment by karlitschek** [comment]
again. having two apps that do similar things also doesn't say anything about default, releases, replacements or possible collaboration or not.  if you look at other platfirms like windows, mac, linux, android, ios and so on then you see that there are more then one application for a task. and this is a good thing not a bad thing. i'm actualy super happy that we have two active and great tasks apps. it's not up to us to decide who contributes where or which user uses what. both should be offered to the users. this is all very good and nothing to decide or do here. great for developers and users to have choice.

**Comment by jancborchardt** [comment]
> if you look at other platfirms like windows, mac, linux, android, ios and so on then you see that there are more then one application for a task

Sure there are more available, offered via an appstore by independent developers. None of these platforms focus on multiple apps for the same thing for the core experience. One Calendar app each, one Contacts app each, etc.
Apart from that the comparison is a bit flawed because all these platforms are way more established than us already. We are not in a position where we should work on 2 Tasks apps just because we think it’s fun. It will result in a sub-par user experience.
https://github.com/owncloud/tasks is the one we focus on, and Tasks+ can stay in its own repo. No need to bring it all in the ownCloud org. 

> i'm actualy super happy that we have two active and great tasks apps

Well, we have to be honest here because it’s about what we offer to the people using our software. One of the tasks apps is clearly more active and greater than the other one.
Focusing on quantity rather than quality and having many apps which have their issues is exactly the reason ownCloud was criticized in the past. We’re improving vastly in this aspect, but I hope you haven’t forgotten that.

> both should be offered to the users.

Of course, this is already done through apps.owncloud.com

**Comment by jancborchardt** [comment]
Apart from this, having the app differently named in 8.2 (or 8.3) will result in the Cal / CardDAV addresses changing and everyone needing to change it manually. This is a big reason for replacing an app properly, whichever we should decide to have as the »main« one. (I’ll not say »Official« anymore.)

**Comment by jospoortvliet** [comment]
Ok, so I think we have a difference here between two rather fundamentally different ideas of what the central ownCloud project is for. Risking representing them wrongly, I think there are two points of view:
- anything done by what is arguably 'ownCloud community member(s)' can go in
- It is for official apps we're collectively responsible for, so only well maintained apps go in

Currently, the situation is probably roughly in between the two different views and neither is documented or clarified anywhere (which, in itself, is not good).

Now I think it is a horrible idea to discuss which of these two points above is The Right One while mixing it up with this particular repo move.

I suggest that we discuss the goal/meaning of the central ownCloud project at the conference. Jan, if you want, you can even book a specific time in the meeting room to ensure it happens: https://github.com/owncloud/core/wiki/Conference-meetings

In the mean time, how about we keep this open. It's just three weeks from now so that won't kill any puppies and after that, we can make an easy decision - as we probably would have to move many other apps out, too, if we'd go for the 2nd option.

As long as the choice is properly explained and argued, I don't think either of the two is really wrong, by the way. But it is something to talk about together.

**Comment by marvin-martian** [comment]
As a user of the apps I thought I would add my 2c to this discussion, for me:

Calendarplus wins hands down in terms of functionality and features that are missing from Calendar, particularily calendar subscriptions.

Calendar has a more simplistic GUI than Calendarplus, and in some ways keeping it simple (or compact) is arguably more desirable from a usability perspective. I find the way Calendarplus splits the settings to the top-right a design faux pas, it should be consistently in the bottom left menu like in most of the other apps. 

The top bar of the Calendarplus however is useful, particularily when viewed from a mobile because it is quicker to navigate between months, something that Calendar lacks.

In my ultimate world the features of Calendarplus need to be merged into the GUI of Calendar. By doing this it would also reduce the necessity of migration of existing calendars and caldav links. 
Considering the lack of developers for maintainability, the focus should be on a single calendar app to not dilute efforts. The same goes for contacts+.  The features need to be merged.

If you ask me Tasks/or Tasks+ could also be merged into the newly merged Calendars. That way it can become a single suite. Working together with the community, owncloud can progress further.

You have the opportunity to take the best ideas/elements from the existing code bases, I believe this will benefit all users.

**Comment by Bullnados** [comment]
One other question, why is planned to get +Apps "official" already for 8.2 and not for 9.0? More time to fix bugs and improve all apps! Also it is a big change perfect for a major release. From 8.x.x to 9.0.0.

**Comment by jancborchardt** [comment]
@Bullnados yeah, maybe that’s actually something we should do. With the smaller cycles it’s very difficult. @DeepDiver1975 @karlitschek what do you think?

> In my ultimate world the features of Calendarplus need to be merged into the GUI of Calendar.

@marvin-martian Yup, that’s also what I’m gunning for. I’m working with @libasys to modify the design of Calendar+ to fit ownCloud more, so we’re on the way. :)

> If you ask me Tasks/or Tasks+ could also be merged into the newly merged Calendars. That way it can become a single suite.

@marvin-martian We shouldn’t merge these apps – except the protocol they are different things, and each app should be atomic.
Otherwise I strongly agree with your points regarding community and app development. How about you join us in #owncloud-dev (on freenode IRC) or in case you’re in Berlin at https://owncloud.org/conf (goes for anyone else here too :)

**Comment by marvin-martian** [comment]
@jancborchardt Unfortunately I am stuck in Zurich so Berlin is out for me. Though I might pop into the IRC channel.

After some more thought about it, a bigger question needs to be answered by ownCloud. 

**How far into groupware should ownCloud go?** 

All this stuff **is** groupware because it allows communication, co-ordination and collaboration. For groupware, in my mind, there are 5 [^1] apps that all need each other, though dependencies vary: Mail, Calendar, Contacts, Tasks and Files.

Mail could utilize Contacts, Files and Calendar [^2] 
Calendar could utilize Contacts, Mail and Files
Tasks could utilize Calendar, Contacts, Mail and Files

If having groupware is desired, then keeping these as "official" core apps would seem to be the right way to go because the dependencies are so intertwined. The API's need to be clear and consistent so they can all hook into each other without having to reinvent the wheel. The developers need to work hand in hand.

What seems to have happened, as these features are not "officially" seamlessly integrated, the community has run off to do it themselves or tried to add the parts that are missing, without having a bigger picture of how they can all work together.. efficiently. The importance of these apps is clear from the community just by the usage and the efforts made by individuals in the community.

This also opens up the subject of app dependency. I read somewhere [^3](https://github.com/owncloud/core/issues/10777#issuecomment-55651194) that this is not wanted in owncloud but maybe it has to be re-thought if you want to attempt creating true groupware.

_Notes:_
[^1]: and possibly a 6th.. Notes 
...and possibly a 7th (but not existing yet) some sort of webRTC app -  like a sip-client.
I am sure there are others.. Planers etc.. 

> @marvin-martian We shouldn’t merge these apps

[^2]: BTW Mozilla just integrated Lightning Calendar into Thunderbird 38 maybe because calendars in mail makes sense. 

_Disclaimer:_  I am not owncloud app developer just a simple user (just a fly on a wall)  but I do have a feel for what kind of goes on behind an application.

Sorry for the long post.

**Comment by karlitschek** [comment]
So this is a big questions. :-)
I can try to answer the core part of the question. So the people who are working on the ownCloud core should and are committed to _enable_ the development of this features and move more into the groupware space. So the core should make it possible that such apps are implemented and work in the best possible way. The ownCloud core should also implement all the needed APIs that needs to be provided and consumed by such groupware apps. Remember that we don't want to have hard dependencies between apps so this should go through the core.

But the real question if ownCloud should and will become a groupware as described is nothing that the core people can answer. It would be cool and welcomed of course but this is nothing can can officially be decided. It 100% depends if someone does the work or not. So if you guys keep on doing your awesome work around this groupware apps then we will have more features in this area. This would be awesome. But if not then probably not.
The old open source principal 'who codes decides' is also true in ownCloud. :-)

**Comment by farson2003** [comment]
1. @jancborchardt commented“ maybe it’s time to move the Calendar+ and Contacts+ repositories over to the ownCloud organization so we can work more effectively on them?”

I agree. Why not start to move just the Calendar+ App. If some of you are hesitant about the other apps, why not just move Calendar+ first? That would facilitate @libasys work and draw more attention and possible supporters towards his app!
1. @DeepDiver1975 wrote: „If so we move all or nothing”. 
   I do not think that has to be the case, or am I not aware of a technical reason?!
2. A suggestion: As soon as Calendar+ meets the criteria to be “official” or “approved” and “not experimental” anymore, this should be marked as such. This again would build up trust and attract more potential supporters/developers. That refers to my issue here: https://github.com/libasys/calendarplus/issues/53

**Comment by deMattin** [comment]
Maybe official calendar (and contacts) is a lack of features - but almost everything simply works really well.
Here in this git actually 80 issues are open (many without comments) and the 20 closed are very small fixes or no issues.
Since one month no new pulls - even not in "rework" branch and (simple to review) pull request are sitting there over a month.
Some features of "official" are missing like adress completition of event adresses and contactsplus is a construction area with big issues but needed by calendarplus and so both have to be seen as a packet.
This all would be no problem, if the project(s) only would develop.

Actually it looks like, if the maintainer has no or not enough time to work on this project, and so it might stuck as other apps and we don't need another marginally cared app in "official" part of owncloud.
And to be honest - without one or more really enthusiastic maintainer with enough time, there's no chance to bring such a big project to a good state.
And it's better to have working apps "official and approved" with less functionality, than a feature rich one which doesn't seem to be maintained and bug fixed enough.

Today I tried to switch to contactsplus and calendarplus, but I will have to give my users a readme, what they don't are allowed to do, to get it all working and so probably I have to roll back to "official apps" with less features but working and "input tolerant".
I like calendarplus and contactsplus and the additional features and I really wished, that this project will go further. But if it develops like actual, it will be a long way to use it in production for all users, that might not be able or want to search for the reason of errors like I did for around 2 days.

And I see it another way round as stated here:
Porting it in official part will not push development but will slow it down because there are many more "core teamers" involved to review and not only the maintainer(s), who can try or simply do (pull), what they think is best and repair bugs and let some others test it.

Sorry for my maybe hard words but this is, how it looks like for me (maybe I'm wrong and @libasys  develops on his own on local git? I don't know.).
But it's not the time to go official with this project now and I also think, this won't push the project.

And as @jancborchardt already said:
`Focusing on quantity rather than quality and having many apps which have their issues is exactly the reason ownCloud was criticized in the past.`

**Comment by farson2003** [comment]
@deMattin wrote "Maybe official calendar  is a lack of features - but almost everything simply works really well.   Here[...] 80 issues are open [...] and the 20 closed are very small fixes or no issues.
Since one month no new pulls [...]"

I agree with the above. Like many admins we are currently setting up an Owncloud instance, and we have to decide between a more advanced Calendar+ and a more stable but less featurerich Calendar[simple].
Version 1.0.3 of Calendar+ with important bugfixes is still not availabe (see https://github.com/libasys/calendarplus/issues/108) and there are a few important issues around.

Initial optimism about Calendar+ gets more blurred and some may now start to advise to install the "old" calendar app when setting up new OC instances, having in mind that for OC instances with many users a migration from Calendar[simple] to Calendar+ will be nearly impossible.

We cannot blame @libasys , as we all see the hard work, but above lines speak for themselves (and @jancborchardt  is right, quality must have priority!).

I will therefore now equally have to advise to stick to the old Calendar, even though it hurts my heart.

**Comment by menelic** [comment]
@farson2003 @jancborchardt as a user and (small-time) admin I am following this debate, sitting on the fence about whether or not to implement the plus suite in my instances. It seems @libasys has implemented some of  @jancborchardt s critical points re: UI and design and adressed some of the other major criticisms in recent releases. Any chance for another attempt at creating synergies so that users can benefit from the more fully featured plus suite as standard maintained by a collaborative team? Both the Plus suite and teh "standard" apps have seen several releases recently - can't this energy be combined in a consolidated plus suite now that many critical points have been adressed?

**Comment by farson2003** [comment]
@menelic wrote "can't this energy be combined in a consolidated plus suite now that many critical points have been adressed?"

I would also wish so!
